File: Chapter4/Python/Example2.py
# Chapter 4 - Poisson Distribution
# Example 2: Call Center Traffic Analysis  Poisson(30)

from scipy.stats import poisson

# Parameter
lam = 30

# i. P(X = 25)
prob_exact = poisson.pmf(25, lam)
print(f"i. P(X = 25) = {prob_exact:.4f}")

# ii. P(X > 35) = 1 - P(X <= 35)
prob_greater = 1 - poisson.cdf(35, lam)
print(f"ii. P(X > 35) = {prob_greater:.4f}")

# iii. 95th percentile
percentile_95 = poisson.ppf(0.95, lam)
print(f"iii. 95th percentile: {int(percentile_95)} calls")

# Verification
print(f"Verification: P(X <= {int(percentile_95)}) = {poisson.cdf(percentile_95, lam):.4f}")

File: Chapter4/R/Example2.R
# Chapter 4 - Poisson Distribution
# Example 2: Call Center Traffic Analysis  Poisson(30)

# Parameter
lambda <- 30

# i. P(X = 25)
prob_exact <- dpois(25, lambda)
cat(sprintf("i. P(X = 25) = %.4f\n", prob_exact))

# ii. P(X > 35) = 1 - P(X <= 35)
prob_greater <- 1 - ppois(35, lambda)
cat(sprintf("ii. P(X > 35) = %.4f\n", prob_greater))

# iii. 95th percentile
percentile_95 <- qpois(0.95, lambda)
cat(sprintf("iii. 95th percentile: %d calls\n", percentile_95))

# Verification
cat(sprintf("Verification: P(X <= %d) = %.4f\n",
            percentile_95, ppois(percentile_95, lambda)))
